### ISSUE TITLE
Changing the number of digits

### ISSUE THREAD

**Issue by holstius** [nicolaskruchten/pivottable#559]
Not quite sure where/how to file this.

Sometimes one is working with quantities spanning several orders of magnitude. Depending on the dataset --- but sometimes just (1) the aggregator, or the (2) level to which one has disaggregated the data --- it can be nice to "bump" the number of **digits** up or down temporarily.

The current design allows for a run-time `digitsAfterDecimal`, bundling that with other aspects of localization. I've found it a bit difficult to reach, but acknowledge that I've coming at it from the land of [rpivotTable](https://github.com/smartinsightsfromdata/rpivotTable) users who know very little about Javascript.

I am not sure if you've considered making it possible to change aspects of the locale on-the-fly, or to separate the linguistic and number-formatting pieces a bit more.

In any case, just having a top-level `digitsAfterDecimal` or `significantDigits` argument to `pivotUI()` would be welcome from where I stand. But you may have thought through the design in a way I'm not seeing, of course. 

Thanks in any case! It's a wonderful tool.

**Comment by nicolaskruchten** [comment]
Unfortunately, number formatting is basically a feature of the locale and probably will remain so... Part of the problem is that the number of digits is what is used to create different aggregators like "sum" and "integer sum", so one can't just have a global tuning knob for "number of digits".

Idea: if we had a "scientific" locale where the number formatter basically returned a fixed-precision output in scientific notation, would that address part of your "on the fly" concern?

**Comment by nicolaskruchten** [comment]
Specifically: we could use the Javascript `toPrecision(n)` function as a number formatter...

```javascript
for(var i=0; i> -15; i--) console.log( (12345*Math.pow(10,i)).toPrecision(4) )
```

returns

```
1.235e+4
1235
123.5
12.35
1.235
0.1235
0.01235
0.001234
0.0001234
0.00001235
0.000001234
1.234e-7
1.234e-8
1.235e-9
1.234e-10
```

**Comment by holstius** [comment]
toPrecision() seems like a great solution. It would definitely cover my use cases!

Thanks for thinking about this!

**Comment by tsando** [comment]
Thanks for pointing out one can use "Integer Sum" to display a value as integer instead of the default number formatting. I agree with @holstius that a top level formatting option would be nice.

**Comment by ckatsulis** [comment]
Is this functionality currently supported in rpiviottables?  I have tried to include a `toPrecision`, `digitsAfterDecimal `, and `significantDigits` options in the renderRpivottable call without success.  If it's not supported, is there an easy way to go in and modify the js code for use within R and Shiny?

Thanks

**Comment by nicolaskruchten** [comment]
@ckatsulis there is no easy way today to control this from rpivotTable, no, unfortunately.

**Comment by ckatsulis** [comment]
How about modifying in the js source?

On Mon, Aug 14, 2017 at 10:02 AM, Nicolas Kruchten <notifications@github.com

**Comment by nicolaskruchten** [comment]
Yes, the JS source in rpivotTable could be modified but then you would sort of be on your own running a forked version, not able to get CRAN updates etc...

**Comment by nicolaskruchten** [comment]
Probably the best thing to do would be for rpivotTable (and the corresponding Jupyter system which I maintain) to use a number formatter with configurable significant digits.

**Comment by ckatsulis** [comment]
Is this a priory for you?  There seems to be some necessity based on the
forum.  If so, what do you think the timeline is for such a change?
Otherwise I will work on an alternative solution.  Regardless, I appreciate
your interest in the thread.

Thanks,

Chris

On Mon, Aug 14, 2017 at 11:07 AM, Nicolas Kruchten <notifications@github.com

**Comment by nicolaskruchten** [comment]
Well, I'm not the maintainer of rpivotTable, so there isn't much I can do to help your specific situation, sorry... I haven't seen this issue be reported on the Python side yet, so it's not really much of a priority just now on that end for me either.

**Comment by smartinsightsfromdata** [comment]
Hi

Apologies but between holidays and all I ended up with even less time as usual.

I will look into relasing a new version sometime at the beginning of next week.

Thank you for your patience.

Enzo

**Comment by holstius** [comment]
@nicolaskruchten Do you have an inkling of where `toPrecision()` needs to fit in? I might take a stab at this if I knew your opinion on that.

**Comment by nicolaskruchten** [comment]
@holstius if you want to have fixed-precision display from within R then it's going to be pretty involved:

1. You'll need to create a number-formatting function using `toPrecision()`
2. You'll need to create a locale file that uses that number-formatting function
3. You'll need to create a version of rpivotTable which uses this locale file
4. You'll need to send a pull-request to the rpivotTable repo to get these changes merged into the official version (optional)

I'm sorry that this is so complex and that I don't have more time to devote to it :(

**Comment by ldfreight** [comment]
I am curious to know how this tutorial works with 2 digits of precision in the example of HairEyeColor dataset, but has 0 digits of precision in the example of cars.

https://cran.r-project.org/web/packages/rpivotTable/vignettes/rpivotTableIntroduction.html

Anyone has an idea on that?

**Comment by SergiyIurenko** [comment]
That is because in case of HairEyeColor aggregator is Sum=same precision as input data and in case of cars aggregator is Count=always int
